ginga/util/iqcalc.py

```diff
@@ -630,10 +630,82 @@ def fwhm_data(self, x, y, data, radius=15, method_name='gaussian'):
         """Equivalent to :meth:`get_fwhm`."""
         return self.get_fwhm(x, y, radius, data, method_name=method_name)
 
+    # Encircled and ensquared energies (EE)
+
+    def ensquared_energy(self, data):
+        """Return an array of ensquared energy at corresponding pixel indices.
+
+        Data is already a masked array and is assumed to be centered.
+
+        """
+        tot = data.sum()
+        ny, nx = data.shape
+        cen_x = int(nx // 2)
+        cen_y = int(ny // 2)
+        ee = []
+
+        if ny > nx:
+            n_max = ny
+            cen = cen_y
+        else:
+            n_max = nx
+            cen = cen_x
+
+        if n_max % 2 == 0:  # Even
+            delta_i1 = -1
+        else:  # Odd
+            delta_i1 = 0
+
+        for i in range(n_max - cen):
+            ix1 = cen_x - i + delta_i1
+            if ix1 < 0:
+                ix1 = 0
+            ix2 = cen_x + i + 1
+            if ix2 > nx:
+                ix2 = nx
+            iy1 = cen_y - i + delta_i1
+            if iy1 < 0:
+                iy1 = 0
+            iy2 = cen_y + i + 1
+            if iy2 > ny:
+                iy2 = ny
+            ee.append(data[iy1:iy2, ix1:ix2].sum() / tot)
+
+        return ee
+
+    # This is adapted from poppy package. See licenses/POPPY_LICENSE.md .
+    def encircled_energy(self, data):
+        """Return an array of encircled energy at corresponding pixel indices.
+
+        Data is already a masked array and is assumed to be centered.
+
+        """
+        y, x = np.indices(data.shape, dtype=float)
+        cen = tuple((i - 1) * 0.5 for i in data.shape[::-1])
+        x -= cen[0]
+        y -= cen[1]
+        r = np.sqrt(x * x + y * y)
+
+        ind = np.argsort(r.flat)
+        sorted_r = r.flat[ind]
+        sorted_data = data.flat[ind]
+
+        # data is already masked
+        csim = sorted_data.cumsum(dtype=float)
+
+        sorted_r_int = sorted_r.astype(int)
+        deltar = sorted_r_int[1:] - sorted_r_int[:-1]  # assume all radii represented
+        rind = np.where(deltar)[0]
+
+        ee = csim[rind] / sorted_data.sum()  # Normalize
+        ee.set_fill_value(0)
+        return ee.filled().tolist()
+
     # EVALUATION ON A FIELD
 
     def evaluate_peaks(self, peaks, data, bright_radius=2, fwhm_radius=15,
-                       fwhm_method='gaussian', cb_fn=None, ev_intr=None):
+                       fwhm_method='gaussian', ee_total_radius=10,
```

Comment:
@ejeschke , are you okay with this API change?

Reply:
Just adding some kwargs, right? No problem.

Reply:
> heoretically, the fitting functions could also be done inside self.iqcalc.evaluate_peaks() above and store as elements in qs but I am not sure if it is worth the extra memory... but it would avoid code duplication in plotting

I think that makes sense.

Reply:
Yes, just a new kwarg. But if you been passing kwarg as positional, this might break something, so just want to make sure. Thanks!

Reply:
I did end up refactoring how the internal works, and passing functions indeed did simplify some things.